### PR TITLE
pipe2 not available on macOS/__APPLE__

### PR DESCRIPTION
**Pull request by kimim**
fix the problem on macOS Sonoma 14.5/Apple M3

```
[  3%] Building C object open62541/CMakeFiles/open62541-plugins.dir/arch/eventloop_posix/eventloop_posix.c.o
/Users/kimim/workspace/opcua-examples/common/c/open62541/arch/eventloop_posix/eventloop_posix.c:171:15: error: call to undeclared function 'pipe2'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    int err = UA_EventLoopPOSIX_pipe(el->selfpipe);
              ^
/Users/kimim/workspace/opcua-examples/common/c/open62541/arch/eventloop_posix/eventloop_posix.h:322:38: note: expanded from macro 'UA_EventLoopPOSIX_pipe'
# define UA_EventLoopPOSIX_pipe(fds) pipe2(fds, O_NONBLOCK)
                                     ^
/Users/kimim/workspace/opcua-examples/common/c/open62541/arch/eventloop_posix/eventloop_posix.c:171:15: note: did you mean 'pipe'?
/Users/kimim/workspace/opcua-examples/common/c/open62541/arch/eventloop_posix/eventloop_posix.h:322:38: note: expanded from macro 'UA_EventLoopPOSIX_pipe'
# define UA_EventLoopPOSIX_pipe(fds) pipe2(fds, O_NONBLOCK)
                                     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/unistd.h:470:6: note: 'pipe' declared here
int      pipe(int [2]);
         ^
1 error generated.
```

### CODE REVIEW

**Comment by jpfr** [comment]
It is also not available on windows.
You can add the macro #ifdef to use the windows workaround also for MacOS.

**Comment by jpfr** [comment]
I don't have a Mac available right now.
But it should work - it is just opening a UDP socket instead of a UNIX socket pair...

**Comment by kimim** [comment]
OK, Let me try again.

**Comment by kimim** [comment]
@jpfr now, socket is used for __APPLE__ also.

**Comment by jpfr** [comment]
Nice, merged.
